Content.Server/StationEvents/Events/MassHallucinationsRule.cs

```diff
@@ -19,27 +19,29 @@ protected override void Started(EntityUid uid, MassHallucinationsRuleComponent c
 
         var query = EntityQueryEnumerator<MindContainerComponent, HumanoidAppearanceComponent>();
         while (query.MoveNext(out var ent, out _, out _))
+        {
             if (!HasComp<ParacusiaComponent>(ent))
-                component.AffectedEntites.Add(ent);
+            {
+                var paracusia = EnsureComp<ParacusiaComponent>(ent);
```

Comment:
I don't think the !HasComp check is needed here, since you're using EnsureComp. EnsureComp has an overload with an out bool arg that you can use to avoid adding it to AffectedEntities multiple times.

Reply:
We don't want to affect the players that have paracusia already on them from the trait.

Reply:
> We don't want to affect the players that have paracusia already on them from the trait.

It returns a bool to know if it was even added.

Reply:
> It returns a bool to know if it was even added. The original code is bad because it's doing double the HasComp checks for no reason when it should be doing either HasComp + AddComp or just the EnsureComp.
Gotcha